### PR TITLE
i#5383 mac a64, part 3: Add drsyms and test configurations

### PR DESCRIPTION
**Pull request by derekbruening**
Adds static libdwarf and libelftc libraries built for Mach-O AArch64. These are not yet fully tested but this is a first pass to enable configuring and building drsyms and components that depend on it such as drcachesim.  This enables using local M1 builds to replicate GA CI OSX build warnings.

Fixes build warnings and issues from things like the oldest OSX supported.

Fixes configuration issues from x86-only tests.  Building the tests does not yet succeed.

Issue: #5383

### CODE REVIEW

**Comment by derekbruening** [comment]
x86-32 failure is client.attach_test `No such process` which started appearing on GA CI recently and is lumped under #5725

File: CMakeLists.txt [hunk 1]
```diff
@@ -1826,15 +1824,18 @@ if (UNIX AND X64)
   find_library(liblz4 lz4)
   if (liblz4)
     message(STATUS "Found liblz4: ${liblz4}")
+    if (APPLE)
+      find_path(LZ4_INCLUDE_DIR lz4.h HINTS /usr/local/include /opt/homebrew/include)
+      find_path(LZ4_LIB_DIR liblz4.a HINTS /usr/local/lib /opt/homebrew/lib)
+      include_directories(${LZ4_INCLUDE_DIR})
```
Comment (abhinav92003) [inline CMakeLists.txt:1830]: If find_path fails, looks like the result will be lz4.h-NOTFOUND (https://cmake.org/cmake/help/latest/command/find_path.html),  and it'll just fail in include_directories?

**Review by abhinav92003 — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (derekbruening) [reply CMakeLists.txt:1830]: Looks like it (though the result will be LZ4_INCLUDE_DIR-NOTFOUND).  I mirrored the libsnappy lines above.  The include_ and link_ can be wrapped in an if(), here and for libsnappy, with a warning if they fail.  I went ahead and did that with a shared function for both of these.  It is weird that we need the lib path one after find_library().

**Comment by derekbruening** [comment]
(Jenkins is down for planned service for a few days)

**Pull request merged**